api/Dockerfile

```diff
@@ -57,46 +50,51 @@ RUN set -eux; \
 ###< doctrine/doctrine-bundle ###
 ###< recipes ###
 
-# Copy development php.ini
-RUN mv "$PHP_INI_DIR/php.ini-production" "$PHP_INI_DIR/php.ini"
 COPY --link docker/php/conf.d/api-platform.ini $PHP_INI_DIR/conf.d/
-COPY --link docker/php/conf.d/api-platform.prod.ini $PHP_INI_DIR/conf.d/
+COPY --link --chmod=755 docker/php/docker-entrypoint.sh /usr/local/bin/docker-entrypoint
+COPY --link --chmod=755 docker/php/migrate-database.sh /usr/local/bin/migrate-database
 
-COPY --link docker/php/php-fpm.d/zz-docker.conf /usr/local/etc/php-fpm.d/zz-docker.conf
-COPY --link docker/php/php-fpm.d/www.conf /usr/local/etc/php-fpm.d/www.conf
+ENTRYPOINT ["docker-entrypoint"]
 
-RUN mkdir -p /var/run/php
+HEALTHCHECK --start-period=60s CMD curl -f http://localhost:2019/metrics || exit 1
+CMD [ "frankenphp", "run", "--config", "/etc/caddy/Caddyfile" ]
 
-COPY --link docker/php/docker-healthcheck.sh /usr/local/bin/docker-healthcheck
-RUN chmod +x /usr/local/bin/docker-healthcheck
+# Dev FrankenPHP image
+FROM frankenphp_base AS frankenphp_dev
+ENV APP_ENV=dev XDEBUG_MODE=off
 
-HEALTHCHECK --interval=10s --timeout=3s --retries=3 CMD ["docker-healthcheck"]
+RUN mv "$PHP_INI_DIR/php.ini-development" "$PHP_INI_DIR/php.ini"
 
-COPY --link docker/php/migrate-database.sh /usr/local/bin/migrate-database
-RUN chmod +x /usr/local/bin/migrate-database
+COPY --link docker/php/conf.d/api-platform.dev.ini $PHP_INI_DIR/conf.d/
+COPY --link docker/caddy/Caddyfile /etc/caddy/Caddyfile
 
-COPY docker/php/docker-entrypoint.sh /usr/local/bin/docker-entrypoint
-RUN chmod +x /usr/local/bin/docker-entrypoint
+# renovate: datasource=github-tags depName=xdebug/xdebug
+ARG XDEBUG_VERSION=3.3.2
+RUN set -eux; \
+	install-php-extensions \
+	xdebug-$XDEBUG_VERSION \
+	pcov \
+	;
 
-ENTRYPOINT ["docker-entrypoint"]
-CMD ["php-fpm"]
+CMD [ "frankenphp", "run", "--config", "/etc/caddy/Caddyfile", "--watch" ]
 
-# https://getcomposer.org/doc/03-cli.md#composer-allow-superuser
-ENV COMPOSER_ALLOW_SUPERUSER=1
-ENV PATH="${PATH}:/root/.composer/vendor/bin"
-ENV COMPOSER_HOME=/tmp/composer
-ENV COMPOSER_CACHE_DIR=/tmp/composer/cache
-RUN mkdir -p /tmp/composer/cache
-RUN chmod ugo+w /tmp/composer/cache
+# Prod FrankenPHP image
+FROM frankenphp_base AS frankenphp_prod
 
-COPY --from=composer:2.7 /usr/bin/composer /usr/bin/composer
+ENV APP_ENV=prod
+#ENV FRANKENPHP_CONFIG="import worker.Caddyfile"
+
+RUN mv "$PHP_INI_DIR/php.ini-production" "$PHP_INI_DIR/php.ini"
+
+COPY --link docker/php/conf.d/api-platform.prod.ini $PHP_INI_DIR/conf.d/
+COPY --link docker/caddy/Caddyfile.prod /etc/caddy/Caddyfile
+#COPY --link docker/caddy/worker.Caddyfile /etc/caddy/worker.Caddyfile
```

Comment:
This would be needed to active [frankenphp worker mode](https://frankenphp.dev/docs/worker/) aka [Symfony runtime](https://symfony.com/doc/current/components/runtime.html).

I didn't want to include this in the same PR but left the code which is necessary to make it easier to find the right spots to activate it. Alternatively I can also remove and then we re-add with a separate PR (if we want this).